core/trino-main/src/main/java/io/trino/operator/TopNOperator.java

```diff
@@ -190,12 +204,18 @@ public void addInput(Page page)
     @Override
     public void finish()
     {
+        isFlushing = false;
```

Comment:
maybe call isFlushing => isPartialFlushing. Otherwise, it is unclear why this should be false when the operator is finished.

Reply:
Done